src/geometries/ExtrudeGeometry.js

```diff
@@ -731,7 +731,13 @@ const WorldUVGenerator = {
 		const d_y = vertices[ indexD * 3 + 1 ];
 		const d_z = vertices[ indexD * 3 + 2 ];
 
-		if ( Math.abs( a_y - b_y ) < 0.01 ) {
+		const vectorInFace = new Vector3( b_x - a_x, b_y - a_y, b_z - a_z );
+		const up = new Vector3( 0, 1, 0 );
+
+		const cos45 = 0.70710678;
+		const isMostlyHorizontal = Math.abs( vectorInFace.normalize().dot( up ) ) < cos45;
+
+		if ( isMostlyHorizontal ) {
```

Comment:
Instead, try something simple, like
```js
if ( Math.abs( a_y - b_y ) < Math.abs( a_x - b_x ) ) {
```

Reply:
Ah right, we only need to think about the 2D slope. Good point!

Reply:
Do you know how to squash commits? This really is a 1-line change.

Reply:
Yup, just not in the habit. Squashed!